src/libos/src/net/socket/berkeley_sockets.rs

```diff
@@ -0,0 +1,14 @@
+use super::*;
+
+// Trait for Berkeley socket API
+// TODO: add more functions
+pub trait BSDSockets {
+    fn bind(&self, addr: SockAddr) -> Result<()>;
```

Comment:
The argument `addr` should be a reference to prevent the caller of this method from losing the ownership of `addr` object. The reasoning applies to the `connect` and `sendto` method.